front/src/applications/operationalStudies/views/ManageTrainSchedule.tsx

```diff
@@ -43,6 +46,30 @@ const ManageTrainSchedule = ({ trainIdToEdit }: ManageTrainScheduleProps) => {
   const origin = useSelector(getOrigin);
   const destination = useSelector(getDestination);
   const pathSteps = useSelector(getPathSteps);
+
+  const pathStepWithPointType = useMemo(
+    () =>
+      pathSteps.map((step, index, array) => {
```

Comment:
Instead of using a map and a compact l133, you can directly use a reduce here. Add in `acc` only the steps that are not null
```suggestion
      pathSteps.reduce((acc, step, index) => {
```

Reply:
The `reduce` war continues 🔫

Reply:
Done